layouts/joomla/form/field/calendar.php

```diff
@@ -80,6 +81,13 @@
 	$attributes['required'] = '';
 }
 
+// Data attributes - data-*
+$dataAttribute = '';
+foreach ($dataAttributes as $key => $attrValue)
```

Comment:
```suggestion

foreach ($dataAttributes as $key => $attrValue)
```

Repeat with other instances.